src/main/java/com/github/hanyaeger/api/engine/entities/entity/sprite/delegates/SpriteAnimationDelegate.java

```diff
@@ -102,10 +105,14 @@ public void setAutoCycle(final long interval) {
      * @param row the row to cycle through (zero-indexed)
      */
     public void setAutoCycle(final long interval, final int row) {
+        if (row >= rows) {
+            String message = String.format(INVALID_ROW_EXCEPTION, row, rows);
+            throw new IllegalArgumentException(message);
```

Comment:
Hmmm. Right now we use YaegerEngineExpections for all exceptions related to Yaeger. Do throwing this IllegalArgumentException breaks that consistency.
But it does feel right to stick to Java Exceptions as much as possible. That will give the students insight standard Java Exceptions, which will be more valuable than it being a YaegerEngineException.

Reply:
I agree with both standpoints, how about throwing a YaegerEngineException with an IllegalArgumentException as cause?

Reply:
That's a nice solution. Good thinking...